content/blog/2025/anchor-position-area.md

```diff
@@ -0,0 +1,170 @@
+containing block that does not map to an element.
+
+This containing block is generated by the grid area that you select using
+`position-area`. For instance, with `position-area: top left`, the new
+containing block starts at the top left corner of the anchor's containing block
+and extends to the top left corner of the anchor itself.
+
+{{ embed.img(
+  src='blog/2025/anchor-area-containing-block.png',
+  alt='A 9-space grid, like a tic-tac-toe board, with an Anchor in the center square, and a square within the top left square for a new containing block.',
+  attrs={
+    'class': 'align-center',
+    'width': '1089',
+    'height': '997'
+  }
+) }}
+
+Note that this is not the positioned element's box, but rather the containing block _for_ the positioned element's box.
+
+## Percentages
+
+One of the ways we observe a containing block is that percentages are resolved
+in relation to the containing block. That means a `padding: 5%` on an element
+positioned using `position-area` will have a padding of 5% of the size of the
+containing block, the section of the grid that was selected.
+
+{{ embed.codepen(
+  id='MYgMpMM',
+  title='position-area containing block',
+  user='jamessw',
+  height=500
+) }}
+
+In this case, the padding on the element positioned with `anchor()` is much
+larger, since its containing block is the viewport, not the grid area created by
+`position-area.
+
+## Self alignment
+
+Using `position-area` also changes the normal `self-alignment`, so that the
+positioned element is placed near to the anchor within its containing block.
+
+What that means is that you can use `place-self` on the positioned element, even
+outside of a `grid` or `flex` context.
+
+{{ embed.codepen(
+  id='raBEmOV',
+  title='position-area self alignment',
+  user='jamessw',
+  height=600
+) }}
+
+## Shifts to prevent overflow
+
+This is perhaps the most magical portion of all. In some instances, the browser
+will adjust the position to attempt to avoid accidental overflow.
+
+If, after applying padding and margins, your positioned element would overflow
+the newly created containing block but would fit inside its original container
+block, it will get shifted to prevent overflow.
+
+In this example, the targets are all positioned with `position-area: center`.
+The top and bottom targets are shifted towards the center because they would
+otherwise overflow their original containing block, which is the `div` with the
+dashed border. They are shifted just enough to be flush with their containing
+block.
+
+{{ embed.codepen(
+  id='XJrLzWa',
+  title='position-area self alignment - absolute positioning',
+  user='jamessw',
+  height=350
+) }}
+
+Note that in that example, the targets are `position: absolute`. If instead, the
+targets are `position: fixed`, the targets do not overflow their original
+containing block (the viewport) and so no shifting happens.
+
+{{ embed.codepen(
```

Comment:
If I recreate the effect of `position-area: center` manually…

```css
  left: anchor(left);
  right: anchor(right);
  top: anchor(top);
  bottom: anchor(bottom);
  place-self: center;
```

…I get the same shifting behavior. So I don't think this is a specific feature of position-area, but a feature of how positioning, self-alignment, and overflow behave generally? Here it is again, without using anchors at all: https://codepen.io/miriamsuzanne/pen/xbKvKPg?editors=1100